src/test/java/de/tum/in/www1/artemis/metis/ConversationIntegrationTest.java

```diff
@@ -428,6 +428,21 @@
         assertThat(unreadMessages).isTrue();
     }
 
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "student1", roles = "USER")
+    void codeOfConduct_isAccepted() throws Exception {
+        var agreement = request.get("/api/courses/" + exampleCourseId + "/code-of-conduct-agreement", HttpStatus.OK, Boolean.class);
+        assertThat(agreement).isFalse();
+    }
+
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "student1", roles = "USER")
+    void codeOfConduct_accept() throws Exception {
+        var agreement = request.postWithResponseBody("/api/courses/" + exampleCourseId + "/code-of-conduct-agreement", HttpStatus.OK, Boolean.class);
+        assertThat(agreement).isTrue();
```

Comment:
I feel like this test is too weak. Can you please add an additional verification that a subsequent call to get the status of the agreement now returns true. 

```java
var agreement = request.postWithResponseBody("/api/courses/" + exampleCourseId + "/code-of-conduct-agreement", HttpStatus.OK, Boolean.class);
        assertThat(agreement).isTrue();

        var newAgreement = request.get("/api/courses/" + exampleCourseId + "/code-of-conduct-agreement", HttpStatus.OK, Boolean.class);
        assertThat(newAgreement).isTrue();
```

Reply:
Agree c1da199fdaf91b5cee8e27664664a154d4f46033